db/db_impl.cc

```diff
@@ -932,7 +944,23 @@ void DBImpl::BackgroundCall2(
   versions_->SetCompactionRunning(level);
 
   if (!shutting_down_.Acquire_Load()) {
-    Status s = BackgroundCompaction(Compact);
+    Status s;
+
+    switch(type)
+    {
+        case kNormalCompaction:
+            s = BackgroundCompaction(Compact);
+            break;
+
+        case kExpiryFileCompaction:
+            s = BackgroundExpiry(Compact);
+            break;
+
+        default:
+            assert(0);
+            break;
+    }   // switch
+
     if (!s.ok() && !shutting_down_.Acquire_Load()) {
       // Wait a little bit before retrying background compaction in
```

Comment:
Should this wait be after any type of compaction or just a background compaction? If background only, then the condition needs to be adjusted; if background or expiry, then the logging statement below should be adjusted to indicate the type of compaction that was attempted.